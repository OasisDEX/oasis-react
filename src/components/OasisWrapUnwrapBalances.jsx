import React, { PureComponent } from "react";
import { PropTypes } from "prop-types";
import ImmutablePropTypes from "react-immutable-proptypes";
import OasisWidgetFrame from "../containers/OasisWidgetFrame";
import { OasisTable } from "./OasisTable";
import OasisInlineTokenBalance from "./OasisInlineTokenBalance";

/* eslint-disable react/prop-types */
const colDefinition = [
  { heading: "coin", key: "unwrappedToken" },
  {
    heading: "wallet",
    // eslint-disable-next-line react/display-name
    template: ({ unwrappedToken, unwrappedBalance }) => (
      <OasisInlineTokenBalance
        inWei={true}
        token={unwrappedToken}
        balance={unwrappedBalance}
      />
    )
  },
  {
    heading: "wrapped",
    // eslint-disable-next-line react/display-name
    template: ({ wrapperToken, wrappedBalance }) => (
      <OasisInlineTokenBalance
        inWei={true}
        token={wrapperToken}
        balance={wrappedBalance}
      />
    )
  }
];

const propTypes = PropTypes && {
  changeRoute: PropTypes.func,
  wrapUnwrapBalances: ImmutablePropTypes.list.isRequired
};
const defaultProps = {};

class OasisWrapUnwrapBalances extends PureComponent {
  constructor(props) {
    super(props);
    this.onTableRowClick = this.onTableRowClick.bind(this);
  }

  onTableRowClick(rowData) {
    const { changeRoute } = this.props;
    changeRoute(`/wrap-unwrap/${rowData.unwrappedToken}`);
  }

  render() {
    const { wrapUnwrapBalances = [] } = this.props;
    return (
      <OasisWidgetFrame heading="BALANCES">
        <OasisTable
          onRowClick={this.onTableRowClick}
          col={colDefinition}
          rows={wrapUnwrapBalances.toJSON()}
        />
      </OasisWidgetFrame>
    );
  }
}

OasisWrapUnwrapBalances.displayName = "OasisWrapUnwrapBalances";
OasisWrapUnwrapBalances.propTypes = propTypes;
OasisWrapUnwrapBalances.defaultProps = defaultProps;
export default OasisWrapUnwrapBalances;
