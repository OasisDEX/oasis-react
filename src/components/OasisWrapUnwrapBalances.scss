@import "./../styles/modules/breakpoints";

.table {
  max-width: 100%;
  th:first-child,
  td:first-child {
    flex-basis: 78px;
    text-align: right;
    @media (max-width: $xxs) {
      flex-basis: 50px;
    }
  }

  th:nth-child(2),
  td:nth-child(2) {
    flex-basis: 128px;
    text-align: right;
    @media (max-width: $xxs) {
      flex-basis: 95px;
    }
  }

  th:nth-child(3),
  td:nth-child(3) {
    flex-basis: 128px;
    text-align: right;
    @media (max-width: $xxs) {
      flex-basis: 100px;
    }
  }

}
