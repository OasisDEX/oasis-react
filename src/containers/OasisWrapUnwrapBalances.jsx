
import React, { PureComponent } from 'react';
import { PropTypes } from 'prop-types';
// import ImmutablePropTypes from 'react-immutable-proptypes';

import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import OasisWrapUnwrapBalances from '../components/OasisWrapUnwrapBalances';
import wrapUnwrap from '../store/selectors/wrapUnwrap';
import platformReducer from '../store/reducers/platform';

const propTypes = PropTypes && {
  actions: PropTypes.object.isRequired
};

export class OasisWrapUnwrapBalancesWrapper extends PureComponent {
  render() {
    const { wrapUnwrapBalances, actions: { changeRoute } } = this.props;
    return (
      <OasisWrapUnwrapBalances
        changeRoute={changeRoute}
        wrapUnwrapBalances={wrapUnwrapBalances}
      />
    );
  }
}

export function mapStateToProps(state) {
  return {
    wrapUnwrapBalances: wrapUnwrap.wrapUnwrapBalances(state)
  };
}
export function mapDispatchToProps(dispatch) {
  const actions = {
    changeRoute: platformReducer.actions.changeRouteEpic
  };
  return { actions: bindActionCreators(actions, dispatch) };
}

OasisWrapUnwrapBalancesWrapper.propTypes = propTypes;
OasisWrapUnwrapBalancesWrapper.displayName = 'OasisWrapUnwrapBalances';
export default connect(mapStateToProps, mapDispatchToProps)(OasisWrapUnwrapBalancesWrapper);
