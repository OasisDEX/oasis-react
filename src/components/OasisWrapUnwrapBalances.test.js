/* global shallow describe it expect */
/* eslint-disable import/first */
import React from 'react';
import { shallow } from 'enzyme';
import OasisWrapUnwrapBalances from './OasisWrapUnwrapBalances';
import { fromJS } from 'immutable';

describe('(Component) OasisWrapUnwrapBalances', () => {
  it('should render', () => {
    const props = {
      wrapUnwrapBalances : fromJS([]),
      resetActiveWrapForm: jest.fn,
      resetActiveUnwrapForm: jest.fn,
    };
    const wrapper = shallow(
      <OasisWrapUnwrapBalances {...props}/>,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
