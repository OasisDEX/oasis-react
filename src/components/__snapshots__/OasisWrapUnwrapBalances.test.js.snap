// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) OasisWrapUnwrapBalances should render 1`] = `
<OasisWidgetFrame
  className=""
  heading="BALANCES"
>
  <OasisTable
    col={
      Array [
        Object {
          "heading": "coin",
          "key": "unwrappedToken",
        },
        Object {
          "heading": "wallet",
          "template": [Function],
        },
        Object {
          "heading": "wrapped",
          "template": [Function],
        },
      ]
    }
    onRowClick={[Function]}
    rows={Array []}
  />
</OasisWidgetFrame>
`;
